code/controllers/subsystems/jobs.dm

```diff
@@ -573,3 +573,132 @@ SUBSYSTEM_DEF(jobs)
+
+/* Procs that alter the save */
+
+//Adds played time to a single, or provided list of jobs. Should only be called automatically.
+/datum/controller/subsystem/job/proc/JobTimeAdd(ckey, jobs, seconds)
+	if(!jobs) return
+	if(!islist(jobs)) jobs = list(jobs)
+	for(var/job in jobs)
+		var/datum/playtime/P = JobTimeGetDatum(ckey, job)
+		if(!P)
+			continue
+		P.playtime += seconds
+		P.realtime += seconds
+		JobTimeSetDatum(ckey, job, P)
+
+
+//Adds a played time to a singular job. Negative numbers subtract time.
+/datum/controller/subsystem/job/proc/JobTimeAlter(ckey, job_key, seconds)
+	var/datum/playtime/P = JobTimeGetDatum(ckey, job_key)
+	P.playtime += seconds
+	JobTimeSetDatum(ckey, job_key, P)
+
+//Sets the played time to a set value
+/datum/controller/subsystem/job/proc/JobTimeSet(ckey, job_key, seconds)
+	var/datum/playtime/P = JobTimeGetDatum(ckey, job_key)
+	P.playtime = seconds
+	JobTimeSetDatum(ckey, job_key, P)
+
+//Forces a job to be accessible regardless of time played, or disables it if 'enable' is set false.
+/datum/controller/subsystem/job/proc/JobTimeForce(ckey, job_key, enable = TRUE)
+	var/datum/playtime/P = JobTimeGetDatum(ckey, job_key)
+	P.forced = enable
+	JobTimeSetDatum(ckey, job_key, P)
+
+/* Procs that read the save */
+
+//Returns true if either the job from job_key has been forced, or the sum of all jobs in the list exceeds req_time
+/datum/controller/subsystem/job/proc/JobTimeAutoCheck(ckey, job_key, jobs, req_time)
+	if(JobTimeAllowCheck(ckey, job_key)) return TRUE
+	return (JobTimeCheck(ckey, jobs) >= req_time)
+
+//Returns the sum of all times in the list of jobs provided.
+//If 'jobs' is not a list, it will be encapsulated in one.
+/datum/controller/subsystem/job/proc/JobTimeCheck(ckey, jobs)
+	var/total = 0
+	if(!islist(jobs)) jobs = list(jobs)
+	for(var/job in jobs)
+		var/datum/playtime/P = JobTimeGetDatum(ckey, job)
+		total += P.playtime
+	return total
+
+
+//Returns whether or not a given job has been forcibly allowed by admins.
+/datum/controller/subsystem/job/proc/JobTimeAllowCheck(ckey, job_key)
+	var/datum/playtime/P = JobTimeGetDatum(ckey, job_key)
+	return P.forced
+
+//Returns an associative list. "job_key" = time
+//Force-states are not considered, only the actual time is returned.
+/datum/controller/subsystem/job/proc/JobTimeGetAll(ckey)
+	//Not Implemented
+
+/* Internal Savefile Stuff - DO NOT CALL FROM OUTSIDE */
+// Seriously, implementation can change wildly depending on our needs.
+
+//We'll only need to read once per key per job, so why do it more?
+/datum/controller/subsystem/job/var/list/playtime_cache = list()
+
+/datum/controller/subsystem/job/proc/JobTimeGetDatum(ckey, job_key, ignore_cache)
+
+	if(playtime_cache[ckey] == null) playtime_cache[ckey] = list()
+
+	var/datum/playtime/P
+	var/cached = TRUE
+	if(!ignore_cache)
+		P = playtime_cache[ckey][job_key]
+	if(!istype(P))
+		var/savefile/S = new("data/player_saves/[copytext(ckey, 1, 2)]/[ckey]/playtime.sav")
```

Comment:
We use SQL server. File IO should not be used for these kinda cases.

Reply:
gotcha it was worth a shot. I did say I had no idea what I was doing